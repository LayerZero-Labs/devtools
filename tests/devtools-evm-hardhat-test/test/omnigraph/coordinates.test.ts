import 'hardhat'
import {
    createContractFactory,
    createProviderFactory,
    getEidForNetworkName,
    getHreByNetworkName,
} from '@layerzerolabs/devtools-evm-hardhat'

describe('omnigraph/coordinates', () => {
    describe('createContractFactory', () => {
        it('should have all contract methods for proxy deployments', async () => {
            const env = await getHreByNetworkName('britney')
            const eid = getEidForNetworkName('britney')

            // Deploy a fixture
            await env.deployments.fixture(['TestProxy'])

            // Now we create a contract factory and observe that the resulting contract has all the contract methods
            const contractFactory = createContractFactory()
            const { contract } = await contractFactory({ contractName: 'TestProxy', eid })

            // We check for existence of the instance method from the actual contract
            expect(contract.contractMethod).toBeInstanceOf(Function)

            // And the existence of the proxy methods
            expect(contract.implementation).toBeInstanceOf(Function)
            expect(contract.functions).toMatchSnapshot()

            // Now let's try and execute a contract method
            const provider = await createProviderFactory()(eid)
            const connectedContract = contract.connect(provider)

            // We execute a contract method
            const contractMethodResult = await connectedContract.contractMethod()
            expect(contractMethodResult.toNumber()).toBe(100)
        })
    })
})
