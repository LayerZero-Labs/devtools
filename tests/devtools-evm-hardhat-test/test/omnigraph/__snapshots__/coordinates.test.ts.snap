// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`omnigraph/coordinates createContractFactory should have all contract methods for proxy deployments 1`] = `
{
  "admin": [Function],
  "admin()": [Function],
  "contractMethod": [Function],
  "contractMethod()": [Function],
  "implementation": [Function],
  "implementation()": [Function],
  "initialize": [Function],
  "initialize()": [Function],
  "upgradeTo": [Function],
  "upgradeTo(address)": [Function],
  "upgradeToAndCall": [Function],
  "upgradeToAndCall(address,bytes)": [Function],
}
`;
